src/libValidator/Validator.cpp

```diff
@@ -226,7 +213,7 @@ bool Validator::CheckCreatedTransactionFromLookup(const Transaction& tx,
     return false;
   }
 
-  if (!VerifyTransaction(tx)) {
+  if (!Transaction::Verify(tx)) {
```

Comment:
Somehow I feel, Validator is the one that validates the txn object and verification should be part of it rather than the Transaction class.

Reply:
The problem is that Validator does too much and validates many things. It gets to a point where in some parts of the code even though only this is needed, it has to depend on the whole of libValidator (which also depends on libMediator, which depends on more and more...).